Replaces notification with panel button

TheYakuzo opened the pull request:
Replaces the dedicated notification component with a more general-purpose panel button.
This allows for more flexible integration of notification features and other possible features in the future.

TheYakuzo committed:
Replaces notification with panel button

Replaces the dedicated notification component with a more general-purpose panel button.
This allows for more flexible integration of notification features.

TheYakuzo committed:
(lint) Cleans up code formatting and improves button control

Addresses minor formatting inconsistencies in the code.

Enhances the MapPanelButton component for better control management
and code readability.

tananaev commented on src/map/components/MapPannelButton.js:
```diff
@@ -1,22 +1,25 @@
 import { useEffect, useMemo } from 'react';
 import { map } from '../core/MapView';
-import './notification.css';
+import './map-pannel-button.css';
```
```suggestion
import './map-button.css';
```

tananaev commented on src/map/components/MapPannelButton.js:
```diff
@@ -1,22 +1,25 @@
 
     this.container = document.createElement('div');
-    this.container.className = 'maplibregl-ctrl-group maplibregl-ctrl';
+    this.container.className = 'maplibregl-ctrl maplibregl-ctrl-group';
```
why this change?

TheYakuzo replied:
useless changes for this. All components have maplibregl-ctrl and after maplibregl-ctrl-group. Triggered me in console

tananaev replied:
OK, let's keep

tananaev commented on src/map/components/MapPannelButton.js:
```diff
@@ -1,22 +1,25 @@
-class NotificationControl {
-  constructor(eventHandler) {
+class PanelButton {
+  constructor(eventHandler, iconClass) {
```
Should we just pass two icons instead of `iconClass` function?

TheYakuzo replied:
So, set the svg directly in component parameter ? 
Or just set the name of the class and I build .maplibre-ctrl-[className]-on / off in component, sound better

tananaev replied:
Actually let's keep the function. I see that it's probably cleaner.

tananaev commented on src/map/components/MapPannelButton.js:
```diff
@@ -27,23 +30,26 @@ class NotificationControl {
 
-const MapNotification = ({ enabled, onClick }) => {
-  const control = useMemo(() => new NotificationControl(onClick), [onClick]);
+const MapPanelButton = ({ enabled, onClick, iconClass }) => {
```
`MapButton`

tananaev replied:
And also the filename.

TheYakuzo replied:
Done

tananaev commented on src/map/components/map-pannel-button.css:
```diff
@@ -0,0 +1,6 @@
+.maplibre-ctrl-icon {
```
Rename the file.

TheYakuzo replied:
Done

tananaev commented on src/main/MainMap.jsx:
```diff
@@ -53,8 +55,13 @@ const MainMap = ({ filteredPositions, selectedPosition, onEventsClick }) => {
+      <MapPanelButton
+        enabled={eventsAvailable}
+        onClick={onEventsClick}
+        iconClass={(status) => `maplibre-ctrl-notification maplibre-ctrl-notification-${status}`}
```
We don't need `maplibre-ctrl-notification` here, right?

TheYakuzo replied:
Yes you're right, don't know why I put it

TheYakuzo commented:
I rework it, let me know if I push :

_// MapButton.js_

```js
import {useEffect, useMemo, useCallback} from "react";
import {map} from "../core/MapView";
import "./map-button.css";

const VALID_TYPES = ["overlay", "notification"];

// Builds the classes for the button based on the type and enabled state
const buildButtonClasses = useCallback((type, enabled = false) => {
	const classes = ["maplibre-ctrl-icon"];

	if (type && VALID_TYPES.includes(type)) {
		classes.push(`maplibre-ctrl-${type}`);
		classes.push(`maplibre-ctrl-${type}-${enabled ? "on" : "off"}`);
	}

	return classes.join(" ");
}, []);

// Dynamically imports the CSS file for the button based on the type
const importCSS = useCallback(async (type) => {
	if (!type || !VALID_TYPES.includes(type)) {
		return;
	}

	try {
		await import(`../${type}/${type}.css`);
	} catch (error) {
		console.error(`Failed to load CSS for ${type}:`, error);
	}
}, []);

class Button {
	constructor(eventHandler, type) {
		if (type && !VALID_TYPES.includes(type)) {
			console.warn(
				`Invalid button type: ${type}. Valid types are: ${VALID_TYPES.join(
					", "
				)}`
			);
		}
		this.eventHandler = eventHandler;
		this.type = type;
	}

	static getDefaultPosition() {
		return "top-right";
	}

	onAdd() {
		this.button = document.createElement("button");
		this.button.className = buildButtonClasses(this.type);
		this.button.type = "button";
		this.button.onclick = () => this.eventHandler(this);

		this.container = document.createElement("div");
		this.container.className = "maplibregl-ctrl maplibregl-ctrl-group";
		this.container.appendChild(this.button);

		return this.container;
	}

	onRemove() {
		this.container.parentNode.removeChild(this.container);
	}

	setEnabled(enabled) {
		this.button.className = buildButtonClasses(this.type, enabled);
	}
}

const MapButton = ({enabled, onClick, type}) => {
	const control = useMemo(() => new Button(onClick, type), [onClick, type]);

	useEffect(() => {
		importCSS(type);
	}, [type, importCSS]);

	useEffect(() => {
		map.addControl(control);
		return () => map.removeControl(control);
	}, [control]);

	useEffect(() => {
		control.setEnabled(enabled);
	}, [enabled, control]);

	return null;
};

export default MapButton;
```

__// MainMap.jsx__
```jsx
 <MapButton
          enabled={overlayEnabled}
          onClick={onOverlayButtonClick}
          type="overlay"
 />
...
<MapButton
          enabled={eventsAvailable}
          onClick={onEventsClick}
          type="notification"
/>
```

tananaev commented:
I don't think we need types validation, but overall I think it's fine to work like that.

tananaev commented:
Actually I don't like the dynamic import. I think we should stick with the original

TheYakuzo committed:
Refactors map panel button to generic map button

Renames and refactors the map panel button to a more generic map button component.

This change aims to improve code reusability and clarity by providing a single, configurable button component for various map controls, instead of having a specific button tailored for panels.

TheYakuzo commented:
Will rebase and use this implementation on the other branch when merged
